pandas/core/indexes/extension.py

```diff
@@ -110,6 +114,15 @@ def wrapper(self, other):
 
 def make_wrapped_arith_op(opname):
     def method(self, other):
+        if (
+            isinstance(other, Index)
+            and is_object_dtype(other.dtype)
```

Comment:
Why only object dtype? You could have Index subclasses with a different dtype?  (or is that not realistic?)

Reply:
That's my suspicion. It's at least true for the xarray case.

@dcherian does xarray have any Index subclasses that have a non `object` dtype?

Reply:
This is a question for @shoyer

Reply:
Xarray only has one Index subclass, which is of object dtype. It's definitely a little fragile, though, so we would like to eventually get rid of it.

Reply:
> so we would like to eventually get rid of it.

IIRC CFTime is for the "UT" timezone, which is always within 1 second of UTC, right?  Can this be represented as a tzinfo object?